Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/Folder.java

```diff
@@ -388,13 +391,21 @@ public void setDpi(Integer dpi) {
     }
 
     /**
-     * Sets the file group of the folder.
+     * Sets the file group of the folder. The file group is the business key of
+     * the folder and may therefore only be set if the object is not yet
+     * persisted ({@code id} is null) or the object is being created
+     * ({@code fileGroup} is null).
      *
      * @param fileGroup
      *            file group to set
      */
     public void setFileGroup(String fileGroup) {
-        this.fileGroup = fileGroup;
+        if (fileGroup.equals(this.fileGroup) || Objects.isNull(this.fileGroup) || Objects.isNull(id)) {
```

Comment:
You are throwing an Exception to the frontend, this will lead to the error page. Please handle the "else" different. Maybe just don't set the fileGroup for now.

Reply:
I don't like this suggestion. This is an unchecked exception. [Unchecked exceptions should be thrown if there is a programming error.](https://stackoverflow.com/a/19061110) At the moment this exception is never thrown because our program behaves correctly. It would be thrown if someone programmed something new and used the object incorrectly. That was the reason why I programmed it that way. Simply not setting the filegroup would lead to data loss and, as a consequence, to behavior that is difficult to understand. I find that awkward. If you do not agree with this, it would be better not to check here at all. If someone changes the value and doesn't know what he's doing, it's his own fault then.

Reply:
so don't check at all